x/concentrated-liquidity/internal/swapstrategy/one_for_zero.go

```diff
@@ -35,33 +34,34 @@ func (s oneForZeroStrategy) GetSqrtTargetPrice(nextTickSqrtPrice sdk.Dec) sdk.De
 	return nextTickSqrtPrice
 }
 
-// ComputeSwapStep calculates the next sqrt price, the new amount remaining, the amount of the token other than remaining given current price, and total fee charge.
+// ComputeSwapStepLegacy is the old implementation of ComputeSwapStepOutGivenIn and ComputeSwapStepInGivenOut.
+// Calculates the next sqrt price, the new amount remaining, the amount of the token other than remaining given current price, and total fee charge.
 //
 // oneForZeroStrategy assumes moving to the right of the current square root price.
 // amountRemaining is the amount of token in when swapping out given in and token out when swapping in given out.
 // amountRemaining is token one.
 // amountOne is token out when swapping in given out and token in when swapping out given in.
 // amountZero is token in when swapping in given out and token out when swapping out given in.
-// TODO: improve tests
-func (s oneForZeroStrategy) ComputeSwapStep(sqrtPriceCurrent, sqrtPriceNextTick, liquidity, amountRemaining sdk.Dec) (sdk.Dec, sdk.Dec, sdk.Dec, sdk.Dec) {
+//
+// To be removed in: https://github.com/osmosis-labs/osmosis/issues/4423
+func (s oneForZeroStrategy) ComputeSwapStepLegacy(sqrtPriceCurrent, sqrtPriceNextTick, liquidity, amountRemaining sdk.Dec) (sdk.Dec, sdk.Dec, sdk.Dec, sdk.Dec) {
```

Comment:
Note: all updates inside this method are done only because we removed `s.isOutGivenIn` flag and use this method only for swap in given out.

Swap out given in is already rewired to the updated `ComputeSwapStepOutGivenIn`. 

The reason why we do not rewire swap in given out is because there are more issues to be addressed in #4423. That will require updating a lot of tests. To limit the scope, this will be done in a separate PR.

Reply:
Due to the reasons above, did not review this method thoroughly  for now